apps/web/components/Dialog.tsx

```diff
@@ -59,12 +59,15 @@ type DialogContentProps = React.ComponentProps<typeof DialogPrimitive["Content"]
 
 export const DialogContent = React.forwardRef<HTMLDivElement, DialogContentProps>(
   ({ children, ...props }, forwardedRef) => (
-    <DialogPrimitive.Content
-      {...props}
-      className="fixed left-1/2 top-1/2 z-50 min-w-[360px] -translate-x-1/2 -translate-y-1/2 rounded bg-white p-6 text-left shadow-xl sm:w-full sm:max-w-lg sm:align-middle"
-      ref={forwardedRef}>
-      {children}
-    </DialogPrimitive.Content>
+    <DialogPrimitive.Portal>
+      <DialogPrimitive.Overlay className="fixed inset-0 z-40 bg-gray-500 bg-opacity-75 transition-opacity" />
+      <DialogPrimitive.Content
+        {...props}
+        className="fixed left-1/2 top-1/2 z-50 min-w-[360px] -translate-x-1/2 -translate-y-1/2 rounded bg-white p-6 text-left shadow-xl sm:w-full sm:max-w-lg sm:align-middle"
+        ref={forwardedRef}>
+        {children}
+      </DialogPrimitive.Content>
+    </DialogPrimitive.Portal>
```

Comment:
Same change coming from #2081 which I merged into the branch. The change here to add the `Dialog.Portal` is to be able to attach the dialog to the `body` and not locally where the tag is used, that gets the dialog to be full page and centered which was the desired behavior.